README.md

```diff
@@ -1,49 +1,214 @@
 
-Happy hacking!
+By default, this command uses Docker and [techdocs-container](https://github.com/backstage/techdocs-container) to make sure all the dependencies are installed. But it can be disabled using `--no-docker` flag.
+
+Command reference -
+
+```bash
+techdocs-cli generate --help
+Usage: techdocs-cli generate|build [options]
+
+Generate TechDocs documentation site using mkdocs.
+
+Options:
+  --source-dir <PATH>             Source directory containing mkdocs.yml and docs/ directory.
+                                  (default: ".")
+  --output-dir <PATH>             Output directory containing generated TechDocs site. (default:
+                                  "./site/")
+  --no-docker                     Do not use docker, use mkdocs executable and plugins in current
+                                  user environment.
+  --techdocs-ref <HOST_TYPE:URL>  The repository hosting documentation source files e.g.
+                                  github:https://ghe.mycompany.net.com/org/repo.
+                                  This value is same as the backstage.io/techdocs-ref annotation
+                                  of the corresponding Backstage entity.
+                                  It is completely fine to skip this as it is only being used to
+                                  set repo_url in mkdocs.yml if not found.
+  -v --verbose                    Enable verbose output. (default: false)
+  -h, --help                      display help for command
+```
+
+### Publish generated TechDocs sites
 
-## Deploying a new version
+```bash
+techdocs-cli publish --publisher-type <awsS3|googleGcs> --storage-name <bucket/container name> --entity <namespace/kind/name>
+```
+
+After generating a TechDocs site using `techdocs-cli generate`, use the publish command to upload the static generated files on a cloud storage
+(AWS/GCS) bucket or (Azure) container which your Backstage app can read from.
+
+The value for `--entity` must be the Backstage entity which the generated TechDocs site belongs to. You can find the values in your Entity's `catalog-info.yaml` file. If namespace is missing in the `catalog-info.yaml`, use `default`.
+The directory structure used in the storage bucket is `namespace/kind/name/<files>`.
+
+Note that the values are case-sensitive. An example for `--entity` is `default/Component/<entityName>`.
 
-Deploying the Node packages to NPM happens automatically on merge to `main` through GitHub Actions. The package is published to `@techdocs/cli` in NPM. Just bump the version number in the package.json file and create a pull request. It will deploy when merged.
+#### Authentication
 
+You need to make sure that your environment is able to authenticate with GCP/AWS. `techdocs-cli` uses the official Node.js clients provided by AWS (v3) and Google Cloud. You can authenticate using
```

Comment:
...And azure

Reply:
PR is up, will get around to reviewing tomorrow. :happytears: This community is so awesome!

https://github.com/backstage/backstage/pull/4017